src/wkt/src/wrappers.rs

```diff
@@ -111,63 +111,98 @@ pub type StringValue = String;
 /// The JSON representation for `BytesValue` is JSON string.
 pub type BytesValue = bytes::Bytes;
 
-impl crate::message::Message for BoolValue {
-    fn typename() -> &'static str {
-        "type.googleapis.com/google.protobuf.BoolValue"
-    }
-    fn to_map(&self) -> Result<crate::message::Map, crate::AnyError>
-    where
-        Self: serde::ser::Serialize + Sized,
-    {
-        let map: crate::message::Map = [
-            (
-                "@type",
-                serde_json::Value::String(Self::typename().to_string()),
-            ),
-            ("value", serde_json::Value::Bool(*self)),
-        ]
-        .into_iter()
-        .map(|(k, v)| (k.to_string(), v))
-        .collect();
-        Ok(map)
-    }
-    fn from_map(map: &crate::message::Map) -> Result<Self, crate::AnyError>
-    where
-        Self: serde::de::DeserializeOwned,
-    {
-        map.get("value")
-            .and_then(|v| v.as_bool())
-            .ok_or_else(crate::message::missing_value_field)
-    }
+macro_rules! impl_message {
+    ($t: ty) => {
+        impl crate::message::Message for $t {
+            fn typename() -> &'static str {
+                concat!("type.googleapis.com/google.protobuf.", stringify!($t))
+            }
+            fn to_map(&self) -> Result<crate::message::Map, crate::AnyError>
+            where
+                Self: serde::ser::Serialize + Sized,
+            {
+                let map: crate::message::Map = [
+                    (
+                        "@type",
+                        serde_json::Value::String(Self::typename().to_string()),
+                    ),
+                    ("value", serde_json::json!(self)),
+                ]
+                .into_iter()
+                .map(|(k, v)| (k.to_string(), v))
+                .collect();
+                Ok(map)
+            }
+            fn from_map(map: &crate::message::Map) -> Result<Self, crate::AnyError>
+            where
+                Self: serde::de::DeserializeOwned,
+            {
+                crate::message::from_value::<Self>(map)
+            }
+        }
+    };
 }
 
+impl_message!(DoubleValue);
+impl_message!(FloatValue);
+impl_message!(Int64Value);
+impl_message!(UInt64Value);
+impl_message!(Int32Value);
+impl_message!(UInt32Value);
+impl_message!(BoolValue);
+impl_message!(StringValue);
+
 #[cfg(test)]
 mod test {
 
     use super::*;
     use crate::Any;
     type Result = std::result::Result<(), Box<dyn std::error::Error>>;
+    use test_case::test_case;
 
-    #[test]
-    fn test_bool_value() -> Result {
-        let input: BoolValue = true;
+    #[test_case(1234.5 as DoubleValue, "DoubleValue")]
+    #[test_case(9876.5 as FloatValue, "FloatValue")]
+    #[test_case(-1234567890123456789 as Int64Value, "Int64Value")]
```

Comment:
I think `int64` and `uint64` values need to be encoded as strings.  `serde_json` tolerates numbers with more bits, but over-the-wire JSON requires these to be stringified.  You may want to split them to a different implementation, like you are (I am guessing) plan to do for `bytes::Bytes`.

Reply:
> I think `int64` and `uint64` values need to be encoded as strings.

Yeah you are correct. Good catch and thanks.

I removed them from this PR. Will add them in a subsequent PR.